src/databricks/labs/dqx/llm/README.md

```diff
@@ -0,0 +1,394 @@
+    print(f"Confidence: {pk_info['confidence']}")
+    
+    # Users can manually create validation rules based on detection results
+    # Example: Create a uniqueness check for the detected primary key
+    # (This would be done separately using the DQX rule creation APIs)
+```
+
+### **Method 4: Direct Detection**
+```python
+from databricks.labs.dqx.profiler.profiler import DQProfiler
+
+profiler = DQProfiler(ws)
+
+# Direct LLM-based primary key detection
+result = profiler.detect_primary_keys_with_llm(
+    table_name="customers",
+    catalog="main",
+    schema="sales",
+    llm=True,  # Explicit LLM enablement required
+    options={
+        "llm_pk_detection_endpoint": "databricks-meta-llama-3-1-8b-instruct"
+    }
+)
+
+if result and result.get("success", False):
+    print(f"✅ Detected PK: {result['primary_key_columns']}")
+    print(f"Confidence: {result['confidence']}")
+    print(f"Reasoning: {result['reasoning']}")
+else:
+    print("❌ Primary key detection failed or returned no results")
+```
+
+## ⚙️ **Configuration Options**
+
+### **LLMConfig Fields**
+```python
+@dataclass
+class LLMConfig:
+    # LLM-based Primary Key Detection (Optional)
+    enable_pk_detection: bool = False  # Enable LLM-based PK detection
+    pk_detection_endpoint: str = "databricks-meta-llama-3-1-8b-instruct"  # LLM endpoint
+    # Note: pk_validate_duplicates is always True and pk_max_retries is fixed to 3
+    # Note: Automatic rule generation has been removed - users must manually create rules
+
+@dataclass
+class ProfilerConfig:
+    # ... other fields ...
+    llm_config: LLMConfig = field(default_factory=LLMConfig)  # LLM configuration
+```
+
+### **Options Parameter Flags**
+```python
+options = {
+    # Simple enablement
+    "llm": True,
+    
+    # Or explicit enablement
+    "enable_llm_pk_detection": True,
+    
+    # Additional LLM options
+    "llm_pk_detection_endpoint": "databricks-meta-llama-3-1-8b-instruct"
+    # Note: llm_pk_validate_duplicates is always True and llm_pk_max_retries is fixed to 3
+    # Note: Automatic rule generation has been removed - users must manually create rules
+}
+```
+
+## 🏗️ **Architecture**
+
+### **Core Components**
+
+1. **`DatabricksPrimaryKeyDetector`** (`pk_identifier.py`)
+   - Main LLM-based detection engine
+   - Handles table metadata retrieval
+   - Manages LLM interactions via DSPy
+   - Performs duplicate validation
+   - Implements retry logic
+
+2. **`DQProfiler`** (Enhanced)
+   - `detect_primary_keys_with_llm()` - Direct LLM detection method
+   - `profile_table_with_llm_pk_detection()` - Convenience method
+   - Lazy LLM imports and graceful fallback
+
+3. **`DQGenerator`** (Enhanced)
+   - `dq_generate_is_primary_key()` - Creates PK validation rules
+   - Handles LLM-specific metadata
+   - Generates traceability information
+
+4. **`ProfilerRunner`** (Enhanced)
+   - Passes LLM configuration options
+   - Orchestrates LLM-enhanced profiling
+
+### **Data Flow**
```

Comment:
I would remove the flow. This is super simple, either you enable pk detection or not. We don't need to describe these flows.